eng/pipelines/runtime-staging.yml

```diff
@@ -234,6 +234,46 @@ jobs:
           eq(variables['monoContainsChange'], true),
           eq(variables['isRollingBuild'], true))
 
+#
+# Build the whole product using Mono and run libraries tests
+#
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/common/global-build-job.yml
+    helixQueuesTemplate: /eng/pipelines/libraries/helix-queues-setup.yml
+    buildConfig: Release
+    runtimeFlavor: mono
+    platforms:
+    - Linux_armv6
+    variables:
+      # map dependencies variables to local variables
+      - name: librariesContainsChange
+        value: $[ dependencies.evaluate_paths.outputs['SetPathVars_libraries.containsChange'] ]
+      - name: monoContainsChange
+        value: $[ dependencies.evaluate_paths.outputs['SetPathVars_mono.containsChange'] ]
+    jobParameters:
+      testScope: innerloop
+      nameSuffix: AllSubsets_Mono
+      buildArgs: -s mono+clr+libs+host+packs+libs.tests -c $(_BuildConfig) /p:ArchiveTests=true
```

Comment:
> clr

This is failing outerloop builds, e.g. https://dev.azure.com/dnceng/public/_build/results?buildId=1569496&view=logs&jobId=86b05f7f-a5b8-51f0-63db-a6d409e0da15&j=86b05f7f-a5b8-51f0-63db-a6d409e0da15&t=53a3521c-9616-50ba-cf1b-e3b67157a105

```
[ 39%] Building ASM object debug/ee/wks/CMakeFiles/cordbee_wks_obj.dir/__/arm/dbghelpers.S.o
<instantiation>:5:9: error: invalid instruction, any one of the following would fix this:
        tst r0, #7
        ^
/__w/1/s/src/coreclr/debug/ee/arm/dbghelpers.S:22:5: note: while in macro instantiation
    CHECK_STACK_ALIGNMENT
    ^
<instantiation>:5:9: note: instruction requires: thumb2
        tst r0, #7
        ^
/__w/1/s/src/coreclr/debug/ee/arm/dbghelpers.S:22:5: note: while in macro instantiation
    CHECK_STACK_ALIGNMENT
    ^
<instantiation>:5:17: note: operand must be a register in range [r0, r7]
        tst r0, #7
                ^
/__w/1/s/src/coreclr/debug/ee/arm/dbghelpers.S:22:5: note: while in macro instantiation
    CHECK_STACK_ALIGNMENT
    ^
<instantiation>:5:9: error: invalid instruction, any one of the following would fix this:
        tst r0, #7
        ^
/__w/1/s/src/coreclr/debug/ee/arm/dbghelpers.S:48:5: note: while in macro instantiation
    CHECK_STACK_ALIGNMENT
    ^
<instantiation>:5:9: note: instruction requires: thumb2
        tst r0, #7
        ^
/__w/1/s/src/coreclr/debug/ee/arm/dbghelpers.S:48:5: note: while in macro instantiation
    CHECK_STACK_ALIGNMENT
    ^
<instantiation>:5:17: note: operand must be a register in range [r0, r7]
        tst r0, #7
                ^
/__w/1/s/src/coreclr/debug/ee/arm/dbghelpers.S:48:5: note: while in macro instantiation
    CHECK_STACK_ALIGNMENT
    ^
```

Reply:
Why?

In `Subsets.props`, `<PrimaryRuntimeFlavor Condition="'$(TargetArchitecture)' == 's390x' or '$(TargetArchitecture)' == 'armv6'">Mono</PrimaryRuntimeFlavor>` should be setting `PrimaryRuntimeFlavor` to `Mono`, then `    <DefaultCoreClrSubsets Condition="'$(PrimaryRuntimeFlavor)' != 'CoreCLR'">clr.iltools+clr.packages</DefaultCoreClrSubsets>` should be setting it so the `clr` subset only builds the supported CoreCLR components

I may be failing to parse how coreclr/ci.yml works.

Reply:
According to the message, this should be `#0-#6`:
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/coreclr/pal/inc/unixasmmacrosarm.inc#L261
Don't know what would be the implication of changing `#7` to `#6`. cc @janvorli

Alternatively, we can skip armv6 here:
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/coreclr/pal/inc/unixasmmacrosarm.inc#L258
```c++
if defined(_DEBUG) && !defined(TARGET_ARMV6)
```
to get by the error.

Reply:
> `Commencing build of "  runtime alljits paltests_install nativeaot spmi " target in "CoreCLR component" for Linux.armv6.Checked in /__w/1/s/artifacts/obj/coreclr/Linux.armv6.Checked`

@directhex hmm... I think it's this template eng/pipelines/coreclr/templates/build-job.yml invoking src/coreclr/build-runtime.sh directly, not using subsets

https://github.com/dotnet/runtime/blob/66a4f235e9f82acdc5b94e49b47be6d873088fb8/eng/pipelines/coreclr/templates/build-job.yml#L199-L202

Reply:
> Don't know what would be the implication of changing #7 to #6

It would be invalid change. 

We should not be compiling this file for arm v6 at all. These .s files would require substantial changes to work correctly on arm v6.

Reply:
I think it is done this way to match s390x coverage (which builds il{d}asm tools along with mono. However, s390x required libunwind and https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/coreclr/pal/inc/unixasmmacross390x.inc to make those work. We would either need armv6 equivalent or disable `clr.iltools+clr.packages` subsets from building armv6. @directhex, per @jkotas' suggestion, we should disable those subsets armv6 for now (that will unblock the CI, outerloop legs).

Reply:
`clr.iltools` should not need any asm macros or libunwind to work. It is accidental dependency.

Reply:
> These .s files would require substantial changes to work correctly on arm v6.

This PR made some assembly changes, those should be revert too? I agree that `clr.iltools` alone does not require many `add_subdirectory()` calls `src/coreclr/CMakeLists.txt` and changes in `jit` etc. which were made for armv6 and s390x shouldn't be needed.

Reply:
A whole bunch of the iltools build fails without asm macros and libunwind, largely via coreclr pal/ and util/ dirs.